app/models/api/commodity.rb

```diff
@@ -48,8 +48,28 @@ def applicable_measures
       @applicable_measures ||= no_additional_code_measures + additional_code_measures
     end
 
+    def applicable_excise_measure_units
+      applicable_measure_units.each_with_object({}) do |(unit, value), acc|
+        measure_sids = value['measure_sids'].select { |measure_sid| measure_sid.in?(excise_measures_sids) }
+
+        next if measure_sids.blank?
+
+        value.deep_dup['measure_sids'] = measure_sids
```

Comment:
I might be missing but this line looks like it does nothing to me - dup the hash, set a key, then ignore the result ?